Xamarin.Forms.Core/Switch.cs

```diff
@@ -39,7 +43,16 @@ public Switch()
 		public bool IsToggled
 		{
 			get { return (bool)GetValue(IsToggledProperty); }
-			set { SetValue(IsToggledProperty, value); }
+			set { SetValue(IsToggledProperty, value); ChangeVisualState(); }
+		}
+		protected internal override void ChangeVisualState()
+		{
+			if (IsEnabled && IsToggled)
+				VisualStateManager.GoToState(this, SwitchOnVisualState);
+			else if (IsEnabled && !IsToggled)
+				VisualStateManager.GoToState(this, SwitchOffVisualState);
+			else
+				base.ChangeVisualState();
```

Comment:
shouldn't you always invoke the base.ChangeVisualState, then handle the specifics ?? /cc @hartez

Reply:
@StephaneDelcroix Yes, you are correct. I retested it and noticed that it will never call 'Normal' state if Switch is enabled but no On/Off states defined by user in Xaml. Hence, I moved it at the top so that it always invoke the base Visual States and then our new On/Off states if available. I've tested various combinations too (added Unit Tests) and it works as expected now (except Focused state - I guess this isn't valid for Switch control and I was unable to set focus on it using `switch.Focus()` ).